### PR TITLE
Import Database from S3 on local dev

### PR DESCRIPTION
**Pull request by chris-moreton**
FYI: The branch is named incorrectly - it should be PAR-235.2.
To review this, you will need to smash everything and reclone, then vagrant up from scratch.
Note in particular the new section in the README file about setting a .env file in your repo root to contain the S3 credentials. Message me for this if you need to.

### CODE REVIEW

**Comment by stemount** [comment]
giving this a whirl

**Comment by stemount** [comment]
How does having this in the `setup.sh` sound to create/copy the env file template if it doesn't exist? A similar mechanism to create the test .env file as the below

`# Setup the development settings file:

if [ ! -f ../web/sites/settings.local.php ]; then
    cp ../web/sites/example.settings.local.php ../web/sites/default/settings.local.php
    cat ../web/sites/settings.local.php.docker.append >> ../web/sites/default/settings.local.php
fi`

**Comment by kalpaitch** [comment]
Talking of this, I'd love to pass this off to a composer post-install hook. Take a look at how Drupal Project handles it... https://github.com/drupal-composer/drupal-project/blob/8.x/scripts/composer/ScriptHandler.php

**Comment by stemount** [comment]
Nice @kalpaitch. I'm blocked from testing this ticket because I don't have an access key for the S3 bucket @chris-moreton

**Comment by kalpaitch** [comment]
Hmmm, fair point @stemount We possibly need to work out how we 'approve' a dev to access S3 buckets. We kinda got to figure it out for Travis still (IAM roles).

**Comment by chris-moreton** [comment]
I don't think we can add the creation of the .env file in the setup.sh as the values inside it are needed as part of the setup.

**Comment by chris-moreton** [comment]
@kalpaitch I think we probably need a development bucket which only copies the latest development data from the test environment.

**Comment by stemount** [comment]
accidentally littered this PR with some other commits, reverted!

**Comment by stemount** [comment]
@chris-moreton have added a few bits, may have gone a bit overboard assuming everybody wants to assign 4 CPUs and 3GB of RAM 😱  - there is a CPU execution cap thing which is probably a lot better.

**Comment by stemount** [comment]
@chris-moreton i realise we now need a third reviewer for this, so I'm swearing in @kalpaitch for this.

File: Vagrantfile [hunk 1]
```diff
@@ -24,9 +24,12 @@ Vagrant.configure("2") do |config|
   config.vm.network "private_network", ip: "192.168.82.68"
   config.vm.network "forwarded_port", guest: 8111, host: 8111
   config.vm.network "forwarded_port", guest: 5411, host: 5411
-  
+
   config.vm.synced_folder ".", "/vagrant", type: "nfs"
   config.vm.provision "shell", inline: $script
 
+  config.vm.provider "virtualbox" do |v|
```
Comment (stemount) [inline Vagrantfile]: this is what may be iffy @kalpaitch, maybe a bit overkill.

**Comment by kalpaitch** [comment]
All looks good apart from the reverted commits, that will most likely impact the PAR-9 branch you're working on @stemount. I'm gonna cherry pick all @chris-moreton's commits to a new branch.

Also I'm not sure about composer manager, what do we need this for? Let's just get the basic PR in...

**Comment by stemount** [comment]
@kalpaitch @chris-moreton lets close this PR then?

**Comment by kalpaitch** [comment]
https://github.com/TransformCore/beis-par-beta/pull/83

**Pull request closed without merging**